src/main/java/com/redhat/management/approval/Logger.java

```diff
@@ -0,0 +1,269 @@
+    hMsg.put("pid", Logger.PID);
+    hMsg.put("tid", String.valueOf(Thread.currentThread().getId()));
+    hMsg.put("level", level);
+    hMsg.put("message", msg);
+    if (httpRequestId.get() != null) {
+      hMsg.put("request_id", httpRequestId.get());
+    }
+
+    ObjectMapper objectMapper = new ObjectMapper();
+    String json = null;
+    try {
+      json = objectMapper.writeValueAsString(hMsg);
+    } catch (JsonProcessingException e) {
+      e.printStackTrace();
+      json = msg;
+    }
+
+    InputLogEvent event = new InputLogEvent();
+    event.setMessage(json);
+    event.setTimestamp(now.toEpochMilli());
+
+    return event;
+  }
+
+  private static String getProcessId() {
+    RuntimeMXBean runtimeBean = ManagementFactory.getRuntimeMXBean();
+    String jvmName = runtimeBean.getName();
+    return jvmName.split("@")[0];
+  }
+
+  public void run() {
+    boolean nonstop = true;
+    while(nonstop) {
+      synchronized (eventQueue) {
+        try {
+          eventQueue.wait();
+        } catch(InterruptedException e) {
+          // should not matter; wait again
+        }
+      }
+
+      // reschedule by canceling the old one and creating a new one
+      if (futureHandle != null && !futureHandle.isDone()) {
+        futureHandle.cancel(false);
+      }
+      futureHandle = this.scheduler.schedule(futureTask, FORCE_FLUSH_INTERVAL, TimeUnit.MILLISECONDS);
+
+      sendLogs(false);
+    }
+  }
+
+  public synchronized void sendLogs(boolean force) {
+    long now = System.currentTimeMillis();
+
+    if(eventQueue.isEmpty()) {
+      lastFlushTime = now;
+      return;
+    }
+
+    if (!force && now - lastFlushTime < FORCE_FLUSH_INTERVAL) {
+      return;
+    }
+
+    List<InputLogEvent> logs = new ArrayList<InputLogEvent>();
+    eventQueue.drainTo(logs);
+
+    for(InputLogEvent log : logs) {
+      System.out.println(log.getMessage());
+    }
+
+    if (client != null) {
+      logToCloud(logs);
+    }
+    lastFlushTime = now;
+  }
+
+  public void setHttpRequestId(String id) {
+    httpRequestId.set(id);
+  }
+
+  private AWSLogsClient initCloudWatchClient() {
+    if (this.accessKeyId == null || this.secretKey == null || this.logGroupName == null || this.logStreamName == null) {
+      return null;
+    }
+    AWSCredentials awsCredentials = new BasicAWSCredentials(accessKeyId, secretKey);
+
+    AWSLogsClient client = (AWSLogsClient) AWSLogsClientBuilder.standard().withCredentials(new AWSStaticCredentialsProvider(awsCredentials)).withRegion(AWS_REGION).build();
+
+    boolean createLogStream = true;
+
+    DescribeLogStreamsRequest request = new DescribeLogStreamsRequest().withLogGroupName(logGroupName).withLogStreamNamePrefix(logStreamName);
+    try {
+      DescribeLogStreamsResult result = client.describeLogStreams(request);
+      for (LogStream stream : result.getLogStreams()) {
+        if (stream.getLogStreamName().equals(logStreamName)) {
+          nextToken = stream.getUploadSequenceToken();
```

Comment:
Not clearly understand about `token`. To me `nextToken` seems not initialized in the first round. Is this expected?

Reply:
`token` is required and maintained by AWS, but it is `null` for the first call in a log stream.